features/upgrade/routing/upgrade.feature

```diff
@@ -208,3 +208,85 @@ Feature: Routing and DNS related scenarios
     Then the step should succeed
     And the output should contain "Hello-OpenShift web-server-rc"
     """
+
+
+  @upgrade-prepare
+  @users=upuser1,upuser2
+  @admin
+  Scenario: Unidling a route work without user intervention - prepare
+    Given I switch to cluster admin pseudo user
+    And I run the :new_project client command with:
+      | project_name | 45955 |
+    Then the step should succeed
+    When I use the "45955" project
+    Given I obtain test data file "routing/web-server-rc.yaml"
+    When I run the :create client command with:
+      | f | web-server-rc.yaml|
+    Then the step should succeed
+
+    And a pod becomes ready with labels:
+      | name=web-server-rc |
+    Then the expression should be true> service('service-unsecure').exists?
+    When I expose the "service-unsecure" service
+    Then the step should succeed
+
+    Given I wait until number of replicas match "1" for replicationController "web-server-rc"
+    Given I have a test-client-pod in the project
+    When I execute on the pod:
+      | curl |
+      | -ksS |
+      | http://<%= route("service-unsecure", service("service-unsecure")).dns(by: user) %>/ |
+    Then the step should succeed
+    And the output should contain "Hello-OpenShift web-server-rc"
+
+    When I run the :idle client command with:
+      | svc_name | service-unsecure |
+    Then the step should succeed
+    Given I wait until number of replicas match "0" for replicationController "web-server-rc"
+
+    # Check the servcie service-unsecure to see the idle annotation
+    When I run the :get admin command with:
+      | resource      | service          |
+      | resource_name | service-unsecure |
+      | n             | 45955            |
+      | o             | yaml             |
+    Then the step should succeed
+    And the output should contain:
+      | idling.alpha.openshift.io/unidle-targets: '[{"kind":"ReplicationController","name":"web-server-rc","replicas":1}]' |
+
+  # @author mjoseph@redhat.com
+  # @case_id OCP-45955
+  @upgrade-check
+  @users=upuser1,upuser2
+  @admin
+  Scenario: Unidling a route work without user intervention
+    # Check the servcie service-unsecure to see the idle annotation is still intact
+    Given I switch to cluster admin pseudo user
+    Given I use the "45955" project
+    When I run the :get admin command with:
```

Comment:
better to use `:get client` instead of `:get admin`

Reply:
done